### ISSUE TITLE
Support Bionic stemcell

### ISSUE THREAD

**Issue by FlorianNachtigall** [cloudfoundry/smb-volume-release#16]
**Describe the bug**

Hi all,

we tried to compile the smb-volume release for Ubuntu Bionic and ran into the following error:
```
+ make
help2man: can't get `--help' info from automake-1.15
Try `--no-discard-stderr' if option outputs to stderr
make: *** [doc/automake-1.15.1] Error 255
```

A possible explanation could be that `automake-1.15` from the `cifs-utils` package no longer works with newer versions of Perl which are consumed from Bionic (see issue: https://github.com/crosstool-ng/crosstool-ng/issues/987#issuecomment-425676468).

**To Reproduce**

```
bosh -e bosh export-release -d compilation-workspace smb-volume/3.1.0 ubuntu-bionic/0.18
```

With deployed manifest:
```
name: compilation-workspace

releases:
- name: smb-volume
  version: "3.1.0"

stemcells:
- alias: default
  os: ubuntu-bionic
  version: latest

instance_groups: []

update:
  canaries: 1
  max_in_flight: 1
  canary_watch_time: 1000-90000
  update_watch_time: 1000-90000
```

**Expected behavior**
Successful compilation and export of tgz packages.

**Version of smb-volume-release**
`3.1.0`

**Full error log:**
```
Task XXXX | 14:03:06 | Compiling packages: cifs-utils/2b5f90c43280309ef547f546893b29386d66c4b4ac3387a6e65aad6c696bade5 (00:01:42)
                      L Error: Action Failed get_task: Task 30846a25-06d1-4396-538f-1da9cee3c248 result: Compiling package cifs-utils: Running packaging script: Running packaging script: Command exited with 2; Truncated stdout:  /usr/bin/install -c -m 644 autoconf.m4 general.m4 status.m4 oldnames.m4 specific.m4 autoheader.m4 autoupdate.m4 autotest.m4 autoscan.m4 lang.m4 c.m4 erlang.m4 fortran.m4 functions.m4 go.m4 headers.m4 types.m4 libs.m4 programs.m4 '/var/vcap/data/compile/cifs-utils/temp/share/autoconf/autoconf'
[...]
Makefile:3687: recipe for target 'doc/automake-1.15.1' failed
, Stderr: + source /var/vcap/packages/python-2.7/bosh/compile.env
++ export PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
++ PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
++ export PYTHONPATH=/var/vcap/packages/python-2.7/lib/python2.7/site-packages:
++ PYTHONPATH=/var/vcap/packages/python-2.7/lib/python2.7/site-packages:
+ temp_path=/var/vcap/data/compile/cifs-utils/temp
+ mkdir /var/vcap/data/compile/cifs-utils/temp
+ tar xf autoconf-2.69.tar.gz
+ pushd autoconf-2.69
+ ./configure --prefix=/var/vcap/data/compile/cifs-utils/temp
+ make
+ make install
+ export PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/var/vcap/data/compile/cifs-utils/temp/bin
+ PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/var/vcap/data/compile/cifs-utils/temp/bin
+ popd
+ tar xf automake-1.15.tar.gz
+ pushd automake-1.15
+ ./configure --prefix=/var/vcap/data/compile/cifs-utils/temp
configure: WARNING: Fortran compiler cannot create executables
configure: WARNING: Fortran 77 compiler cannot create executables
+ make
help2man: can't get `--help' info from automake-1.15
Try `--no-discard-stderr' if option outputs to stderr
make: *** [doc/automake-1.15.1] Error 255
```

Thanks!
cc @stefanwutz

**Comment by amhuber** [comment]
It looks like this should simply require updating the version of automake packaged in the release to 1.15.1 which should be pretty trivial, is there a timeline for this?  This will be a deployment blocker for moving to Ubuntu 18.04 which will be critical shortly for open source CF.

**Comment by amhuber** [comment]
I've confirmed that switching to the new version of automake from https://ftp.gnu.org/gnu/automake/automake-1.15.1.tar.gz does fix the issue.  Creating a dev release with the following changes compiles successfully on bionic and the broker and driver successfully run on the new stemcells.  Here are the changes:

```
diff --git a/config/blobs.yml b/config/blobs.yml
index ad495b8..244b1c4 100644
--- a/config/blobs.yml
+++ b/config/blobs.yml
@@ -2,10 +2,9 @@ autoconf-2.69.tar.gz:
   size: 1927468
   object_id: f26ccc89-cac6-4ab2-42c6-394b6d8911af
   sha: 562471cbcb0dd0fa42a76665acf0dbb68479b78a
-automake-1.15.tar.gz:
-  size: 2244206
-  object_id: c36c457b-640d-4f34-73dd-da239db8d2fd
-  sha: b5a840c7ec4321e78fdc9472e476263fa6614ca1
+automake-1.15.1.tar.gz:
+  size: 2265200
+  sha: sha256:988e32527abe052307d21c8ca000aa238b914df363a617e38f4fb89f5abf6260
 cifs-utils-6.7.tar.bz2:
   size: 363647
   object_id: f76fd32c-1c75-4c28-49fd-48ba34a43204
diff --git a/packages/cifs-utils/packaging b/packages/cifs-utils/packaging
index 0a87316..4cfcbef 100644
--- a/packages/cifs-utils/packaging
+++ b/packages/cifs-utils/packaging
@@ -14,9 +14,9 @@ pushd autoconf-2.69
   export PATH=${PATH}:${temp_path}/bin
 popd

-tar xf automake-1.15.tar.gz
+tar xf automake-1.15.1.tar.gz

-pushd automake-1.15
+pushd automake-1.15.1
   ./configure --prefix=${temp_path}
   make
   make install
diff --git a/packages/cifs-utils/spec b/packages/cifs-utils/spec
index 7f622e7..2b4c05a 100644
--- a/packages/cifs-utils/spec
+++ b/packages/cifs-utils/spec
@@ -6,8 +6,8 @@ dependencies:

 files:
   - autoconf-2.69.tar.gz
-  - automake-1.15.tar.gz
+  - automake-1.15.1.tar.gz
   - libtool-2.4.6.tar.gz
   - talloc-2.1.9.tar.gz
   - pkg-config-0.29.2.tar.gz
```

I'm not sure how to submit a pull request for this given that one of the blobs needs to change.  For now we'll use the dev release we created to unblock our bionic upgrades, but it would be appreciated if a new final release could be created to fix this.

**Comment by amhuber** [comment]
We discovered a new issue related to Bionic stemcells for this release due to https://bugs.launchpad.net/ubuntu/+source/cifs-utils/+bug/1772148.  The issue is resolved by installing the keyutils package on the cells which will need to be included in the smb-volume-release for use on Bionic.

**Comment by JamesClonk** [comment]
Any news on this? We are hitting the same problem too:
```
...
Task 965392 | 15:35:04 | Compiling packages: cifs-utils/2b5f90c43280309ef547f546893b29386d66c4b4ac3387a6e65aad6c696bade5 (00:00:33)
                       L Error: Action Failed get_task: Task 633e8149-d59a-42b7-6a1f-bec0fe0c9cc7 result: Compiling package cifs-utils: Running packaging script: Running packaging script: Command exited with 2; Truncated stdout:  /usr/bin/install -c -m 644 autoconf.m4 general.m4 status.m4 oldnames.m4 specific.m4 autoheader.m4 autoupdate.m4 autotest.m4 autoscan.m4 lang.m4 c.m4 erlang.m4 fortran.m4 functions.m4 go.m4 headers.m4 types.m4 libs.m4 programs.m4 '/var/vcap/data/compile/cifs-utils/temp/share/autoconf/autoconf'
...
Makefile:3687: recipe for target 'doc/automake-1.15.1' failed
, Stderr: + source /var/vcap/packages/python-2.7/bosh/compile.env
++ export PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
++ PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
++ export PYTHONPATH=/var/vcap/packages/python-2.7/lib/python2.7/site-packages:
++ PYTHONPATH=/var/vcap/packages/python-2.7/lib/python2.7/site-packages:
+ temp_path=/var/vcap/data/compile/cifs-utils/temp
+ mkdir /var/vcap/data/compile/cifs-utils/temp
+ tar xf autoconf-2.69.tar.gz
+ pushd autoconf-2.69
+ ./configure --prefix=/var/vcap/data/compile/cifs-utils/temp
+ make
+ make install
+ export PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/var/vcap/data/compile/cifs-utils/temp/bin
+ PATH=/var/vcap/packages/python-2.7/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/var/vcap/data/compile/cifs-utils/temp/bin
+ popd
+ tar xf automake-1.15.tar.gz
+ pushd automake-1.15
+ ./configure --prefix=/var/vcap/data/compile/cifs-utils/temp
configure: WARNING: Fortran compiler cannot create executables
configure: WARNING: Fortran 77 compiler cannot create executables
+ make
help2man: can't get `--help' info from automake-1.15
Try `--no-discard-stderr' if option outputs to stderr
make: *** [doc/automake-1.15.1] Error 255
```
using ubuntu-bionic `1.25` stemcell, smb-volume-release `3.1.0`

**Comment by amhuber** [comment]
We're still using the dev release we created along with a custom release that does an "apt-get install -y keyutils" in a pre-start script.  It works, but it sure would be nice to get this into an updated release.

**Comment by hammondrm** [comment]
Has there been any movement on this issue? Is there a viable workaround, other than creating our own custom release?

**Comment by pivotal-marcela-campo** [comment]
Hi @hammondrm, we are looking into prioritising this work. In the meantime if you need this urgently submitting a PR would definitely help.

**Comment by totherme** [comment]
We have a story [in tracker](https://www.pivotaltracker.com/story/show/181714428) to help us prioritise this. We'll keep the engineering discussion in the open here.

**Comment by fejnartal** [comment]
A new patch (3.11) has been released and is also available in bosh.io.
This patch only includes the fix for `automake`.
According to some comments on this issue there is another thing that needs fixing `keyutils`.
We are actively analysing that issue and will come up with a solution asap.

**Comment by metron2** [comment]
Why is smb-volume so different from nfs-volume? It just uses the packages for nfs-common while this one compiles them.

**Comment by fejnartal** [comment]
Hello @metron2 

I guess you are looking at cifs-utils package and questioning why this releases doesn't use a prepackaged deb for it.
They just seem to have evolved differently.
They also seem to have had different developer communities.

- [nfs-volume-release also compiled its packages at the beginning](https://github.com/cloudfoundry/nfs-volume-release/tree/128577326620fe9ee349079aaae21be9470837ff)
- [But it was later changed to just copy the required files from the compilation vm](https://github.com/cloudfoundry/nfs-volume-release/commit/681bb7069708964667e4b150d6b21cd597381038)
- [And some time later, a different developer added some precompiled debs to the mix](https://github.com/cloudfoundry/nfs-volume-release/commit/cac60e7da182a0ac6312462db10b86451181817c)

On the other hand:
- [smb-volume-release cifs-utils has remained mostly unchanged since the very beginning of the repo](https://github.com/cloudfoundry/smb-volume-release/commits/master/packages/cifs-utils)

**Comment by totherme** [comment]
We just released [3.1.2](https://github.com/cloudfoundry/smb-volume-release/releases/tag/v3.1.2) which contains the keyutils fix.

I think this fixes this issue, so I'm closing. Thanks everyone!
